deploy.pl

```diff
@@ -36,15 +38,55 @@
 
 my $ua = Mojo::UserAgent->new;
 
+my $application_id = $marathon_json->{id};
+
 my $app_url = URI->new("$marathon_apps_url/".uri_escape($marathon_json->{id}))->canonical->as_string();
+
 my $res = $ua->put($app_url => json => $marathon_json)->res();
 
 if ($res->code != 200 && $res->code != 201) {
     die $res->to_string();
 }
 
+my $deployment_url = "$marathon_api_url/deployments";
+
+my $number_of_deployments = number_of_deployments($ua, $deployment_url, $application_id);
+my $deployment_check_wait_time = 5;
+while ($number_of_deployments > 0 && $deployment_verification_timeout > 0) 
+{
+	print "Waiting for $number_of_deployments deployment(s) to finish..." . "\n";	
+	sleep($deployment_check_wait_time);
+	
+	$deployment_verification_timeout -= $deployment_check_wait_time;
+	$number_of_deployments = number_of_deployments($ua, $deployment_url, $application_id);
+}
+
+if ($number_of_deployments > 0) 
+{
+	die "Deployment did not finish successfully. There are still $number_of_deployments ongoing deployment(s). You have to solve this manually in Marathon.";
+}
+else 
+{
+	print "Deployment was successfull.";
```

Comment:
lol, thanks ;-) you see, you can't go to the same english lesson as I do - we are on different levels! :-)